Sources/SmileID/Classes/SelfieCapture/SelfieViewModel.swift

```diff
@@ -314,20 +305,53 @@ public class SelfieViewModel: ObservableObject, ARKitSmileDelegate {
                     )
                 }
                 let authResponse = try await SmileID.api.authenticate(request: authRequest).async()
-                let prepUploadRequest = PrepUploadRequest(
-                    partnerParams: authResponse.partnerParams.copy(extras: extraPartnerParams),
-                    allowNewEnroll: String(allowNewEnroll),
-                    timestamp: authResponse.timestamp,
-                    signature: authResponse.signature
-                )
-                let prepUploadResponse = try await SmileID.api.prepUpload(
-                    request: prepUploadRequest
-                ).async()
-                _ = try await SmileID.api.upload(
-                    zip: zip,
-                    to: prepUploadResponse.uploadUrl
-                ).async()
-                didSubmitSmartSelfieJob = true
+
+                var imageInfoArray = [MultiPartMedia]()
+                if let selfie = try? Data(contentsOf: selfieImage), let media = MultiPartMedia(
+                    withImage: selfie,
+                    forKey: selfieImage.lastPathComponent,
+                    forName: selfieImage.lastPathComponent
+                ) {
+                    imageInfoArray.append(media)
+                }
+                if !livenessImages.isEmpty {
+                    let livenessImageInfos = livenessImages.compactMap { liveness -> MultiPartMedia? in
+                        if let data = try? Data(contentsOf: liveness) {
+                            return MultiPartMedia(
+                                withImage: data,
+                                forKey: liveness.lastPathComponent,
+                                forName: liveness.lastPathComponent
+                            )
+                        }
+                        return nil
+                    }
+
+                    imageInfoArray.append(contentsOf: livenessImageInfos.compactMap { $0 })
+                }
+                let response = if allowNewEnroll {
+                    try await SmileID.api.doSmartSelfieEnrollment(
+                        signature: authResponse.signature,
+                        timestamp: authResponse.signature,
+                        request: MultiPartRequest(
+                            multiPartMedia: imageInfoArray,
+                            userId: userId,
+                            allowNewEnroll: allowNewEnroll,
+                            partnerParams: extraPartnerParams
+                        )
+                    ).async()
+                } else {
+                    try await SmileID.api.doSmartSelfieEnrollment(
```

Comment:
this else branch looks identical to the previous branch

Reply:
good catch, copy paste mistake